py/desispec/test/test_bootcalib.py

```diff
@@ -21,15 +22,18 @@ class TestBoot(unittest.TestCase):
 
     @classmethod
     def setUpClass(cls):
+        cls.origdir = os.getcwd()
+        cls.testdir = tempfile.mkdtemp()
```

Comment:
I think we should also do `shutil.rmtree(cls.testdir)` in `tearDownClass`. `mkdtemp` doesn't clean up after itself, and it does not guarantee that the temporary directory is written to storage that disappears *e.g.* on reboot. In principle, enough tests could fill up a storage device.

Reply:
Thanks.  I agree with that in principle, but in practice I'm a bit nervous that someone in the future might do something like change the `testdir` to some place like $SCRATCH, without realizing that `tearDownClass` will recursively remove the entire directory at the end.  It would be safer to only remove files that are expected to have been generated by the tests, and then `os.rmdir(cls.testdir)` which will fail if the directory isn't empty.  But that could be a maintenance headache maintaining the list of files that each test could generate.  Some of our tests do a brute force `shutil.rmtree` of a tempdir, while others individually remove files (usually because they thought they were otherwise leaving them in the git clone).

I'm going to sit on this for the evening to think about safety vs. maintainability.  Followup thoughts/suggestion welcome.

Reply:
That's a valid point. However you decide to do the clean-up is fine with me. It's more important to clean up than to clean up in a specific way.

Reply:
PS, but it would be nice to have a _consistent_ clean-up method for all tests.

Reply:
I think the best solution is for `tearDownClass` to check that `cls.testdir` starts with `tempfile.gettempdir()` (the base directory used for creating temp subdirectories) before doing a `shutil.rmtree` on it.  That guarantees that the test dir actually is a temporary directory and not something like $SCRATCH.